jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ConditionalHandler.java

```diff
@@ -0,0 +1,659 @@
+     * @param patterns {@link InetAddressPattern}s that are
+     *              tested against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     *              {@link Request#getConnectionMetaData()}.
+     */
+    public void include(InetAddressPattern... patterns)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        for (InetAddressPattern p : patterns)
+            _predicates.include(new InetAddressPredicate(p));
+    }
+
+    /**
+     * Include {@link InetAddressPattern}s in the conditions to be met
+     * @param patterns String representations of {@link InetAddressPattern}s that are
+     *              tested against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     *              {@link Request#getConnectionMetaData()}.
+     */
+    public void includeInetAddress(String... patterns)
+    {
+        for (String p : patterns)
+            include(InetAddressPattern.from(p));
+    }
+
+    /**
+     * Exclude {@link InetAddressPattern}s in the conditions to be met
+     * @param patterns {@link InetAddressPattern}s that are
+     *              tested against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     *              {@link Request#getConnectionMetaData()}.
+     */
+    public void exclude(InetAddressPattern... patterns)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        for (InetAddressPattern p : patterns)
+            _predicates.exclude(new InetAddressPredicate(p));
+    }
+
+    /**
+     * Exclude {@link InetAddressPattern} in the conditions to be met
+     * @param patterns String representations of {@link InetAddressPattern}s that are
+     *              tested against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     *              {@link Request#getConnectionMetaData()}.
+     */
+    public void excludeInetAddress(String... patterns)
+    {
+        for (String p : patterns)
+            exclude(InetAddressPattern.from(p));
+    }
+
+    /**
+     * {@link IncludeExclude#include(Object) Include} a {@link ConnectorAddrMethodPathPredicate}.
+     * If only a method or pathSpec is passed, then this method optimizes by converting this
+     * call to {@link #includeMethod(String...)} or {@link #include(PathSpec...)}.
+     *
+     * @param connectorName optional name of a connector or {@code null}.
+     * @param addressPattern optional InetAddress pattern to test against the remote address or {@code null}.
+     * @param method optional method or {@code null}.
+     * @param pathSpec optional pathSpec to test against the {@link Request#getPathInContext(Request) pathInContext} or {@code null}.
+     */
+    public void include(String connectorName, String addressPattern, String method, PathSpec pathSpec)
```

Comment:
A super specific idea for a small handful of cases.
Why can't the Predicate system alone handle this?

Reply:
It is a convenience method on top of `ConnectorAddrMethodPathPredicate`.  It is not done externally as it has the optimisation of devolving to the more efficient `includeMethod` or `includePath` methods when possible (see javadoc).

This is exactly the use-case we use in the existing InetAccessHandler, so it is not that unusual.

Reply:
Now not needed with decomposed predicates